tests/cupy_tests/indexing_tests/test_generate.py

```diff
@@ -289,3 +289,22 @@ def test_invalid_mode(self, dtype):
             a = tuple([xp.arange(min(dims), dtype=dtype) for d in dims])
             with pytest.raises(ValueError):
                 xp.ravel_multi_index(a, dims, mode='invalid')
+
+
+@testing.gpu
+class TestMaskIndices:
+
+    @testing.numpy_cupy_array_equal()
+    def test_mask_indices(self, xp):
```

Comment:
Note: I tried to parametrize this test over all dtypes, but I believe `cupyx.sparse` (which is  used to generate the data for this test) supports only float and complex types. Is it worth parametrizing over dtypes and generating the input data in a different way?

Reply:
Thanks! I used your idea  of a boolean array as a multiplier to generate a square random matrix of 50% density (exactly  what we were doing before)